packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -0,0 +1,242 @@
+                    className={get(
+                        this.props,
+                        "managedClasses.carousel_flipperPrevious",
+                        ""
+                    )}
+                />
+                <Tabs
+                    label={this.props.label}
+                    activeId={this.state.activeId}
+                    onUpdate={this.handleUpdate}
+                    items={this.slides}
+                    managedClasses={this.generateTabsClassNames()}
+                />
+                <Flipper
+                    direction={FlipperDirection.next}
+                    onClick={this.handleNextClick}
+                    className={get(this.props, "managedClasses.carousel_flipperNext", "")}
+                />
+            </div>
+        );
+    }
+
+    /**
+     * Generate class names
+     */
+    protected generateClassNames(): string {
+        let className: string = get(this.props, "managedClasses.carousel", "");
+        let theme: string = "";
+
+        if (this.getSlideTheme()) {
+            theme =
+                this.getSlideTheme() === SlideTheme.light
+                    ? get(this.props, "managedClasses.carousel__themeLight", "")
+                    : get(this.props, "managedClasses.carousel__themeDark", "");
+
+            className += ` ${theme}`;
+        }
+
+        return super.generateClassNames(className);
+    }
+
+    /**
+     * Returns tabs managedclasses with new carousel-specific JSS
+     */
+    protected generateTabsClassNames(): TabsClassNameContract {
+        return {
+            tabs: get(this.props, "managedClasses.carousel_slides", ""),
+            tabs_tabPanels: get(this.props, "managedClasses.carousel_tabPanels", ""),
+            tabs_tabList: get(
+                this.props,
+                "managedClasses.carousel_sequenceIndicators",
+                ""
+            ),
+            tabs_tabPanelContent: get(
+                this.props,
+                "managedClasses.carousel_tabPanelContent",
+                ""
+            ),
+            tab: get(this.props, "managedClasses.carousel_sequenceIndicator", ""),
+            tab__active: get(
+                this.props,
+                "managedClasses.carousel_sequenceIndicator__active",
+                ""
+            ),
+            tabPanel: get(this.props, "managedClasses.carousel_tabPanel", ""),
+            tabPanel__hidden: get(
+                this.props,
+                "managedClasses.carousel_tabPanel__hidden",
+                ""
+            ),
+        };
+    }
+
+    /**
+     * Get the active slide index
+     */
+    private getActiveIndex(): number {
+        return this.slides.map((slide: Slide) => slide.id).indexOf(this.state.activeId);
+    }
+
+    /**
+     * Get the current slide theme (needed for UI overlay contrast)
+     */
+    private getSlideTheme(): string {
+        if (!this.slides || this.slides.length <= 0) {
```

Comment:
If you make my proposed change, you can change this to just check length because it will *always* be an array